lib/cadence/workflow/executor.rb

```diff
@@ -32,13 +34,33 @@ def run
         return state_manager.decisions
       end
 
+      # Process queries using the pre-registered query handlers
+      #
+      # @note this method is expected to be executed after the history has
+      #   been fully replayed (by invoking the #run method)
+      #
+      # @param queries [Hash<String, Cadence::Workflow::TaskProcessor::Query>]
+      #
+      # @return [Hash<String, Cadence::Workflow::QueryResult>]
+      def process_queries(queries = {})
+        queries.transform_values(&method(:process_query))
+      end
+
       private
 
-      attr_reader :workflow_class, :dispatcher, :state_manager, :metadata, :history, :config
+      attr_reader :workflow_class, :dispatcher, :state_manager, :metadata, :history, :config, :query_registry
+
+      def process_query(query)
+        result = query_registry.handle(query.query_type, query.query_args)
+
+        QueryResult.answer(result)
+      rescue StandardError => error
+        QueryResult.failure(error)
+      end
 
       def execute_workflow(input, workflow_started_event_attributes)
         metadata = generate_workflow_metadata_from(workflow_started_event_attributes)
-        context = Workflow::Context.new(state_manager, dispatcher, metadata, config)
+        context = Workflow::Context.new(state_manager, dispatcher, metadata, config, query_registry)
 
         Fiber.new do
           workflow_class.execute_in_context(context, input)
```

Comment:
Looks like the changes from `lib/temporal/workflow/task_processor.rb` are missing. You can sync those to `lib/cadence/workflow/decision_task_processor.rb`.